src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -951,14 +951,14 @@ final class TrieMap[K, V] private (r: AnyRef, rtupd: AtomicReferenceFieldUpdater
     if (nonReadOnly) readOnlySnapshot().keySet
     else super.keySet
   }
-  override def filterKeys(p: K => Boolean): collection.MapView[K, V] = {
-    if (nonReadOnly) readOnlySnapshot().filterKeys(p)
-    else super.filterKeys(p)
-  }
-  override def mapValues[W](f: V => W): collection.MapView[K, W] = {
-    if (nonReadOnly) readOnlySnapshot().mapValues(f)
-    else super.mapValues(f)
-  }
+
+  override def view: MapView[K, V] = if (nonReadOnly) readOnlySnapshot().view else super.view
```

Comment:
Good catch!

Reply:
All thanks too the existing exhaustive tests :-)